.github/workflows/release.yml

```diff
@@ -1,13 +1,35 @@
-name: Release
+name: CI
+
 on:
   push:
     branches:
       - master
+  pull_request:
+    types: [opened, synchronize]
 
 jobs:
+  test:
+    strategy:
+      matrix:
+        node-version: [10.x, 12.x]
+        os: [ubuntu-latest, macOS, windows-latest]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@master
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node-version }}
+      - run: npm test
+
   release:
     name: release
     runs-on: ubuntu-latest
+    needs: test
+    if: |
+      github.event_name == 'push'
+      && github.ref == 'refs/heads/master'
+      && github.repository == 'probot/create-probot-app'
     steps:
       - uses: actions/checkout@master
       - uses: actions/setup-node@v1
```

Comment:
Thanks for getting this going Maxim! 

I would suggest we create a separate `.github/workflows/test.yml` file for the tests, similar to what I did in most of the Octokit repositories: https://github.com/octokit/core.js/tree/master/.github/workflows.

I don't think we need to run tests before releasing from master, the PR workflow is already enforced, so tests are guaranteed to be green on master `push` events.

What do you think?

Reply:
That will work.